include/IFileSystem.h

```diff
@@ -239,6 +241,37 @@ class IFileSystem : public virtual core::IReferenceCounted
 	\return True if file exists, and false if it does not exist or an error occured. */
 	virtual bool existFile(const path& filename) const =0;
 
+
+	template<typename StringUniqueType>
+	inline core::smart_refctd_ptr<asset::ICPUBuffer> loadBuiltinData()
+	{
+#ifdef _IRR_EMBED_BUILTIN_RESOURCES_
+		std::pair<const uint8_t*, size_t> found = irr::builtin::get_resource<StringUniqueType>();
+		if (found.first && found.second)
+		{
+			auto returnValue = core::make_smart_refctd_ptr<asset::ICPUBuffer>(found.second);	//return core::make_smart_refctd_ptr<asset::CCustomAllocatorCPUBuffer<core::null_allocator<uint8_t> > >(found.second, found.first, core::adopt_memory);
+			memcpy(returnValue->getPointer(), found.first, returnValue->getSize());
+			return returnValue;
+		}
+		return nullptr;
+#else
+		//in one case, IRR_CORE_UNIQUE_STRING_LITERAL_TYPE value returns an incorrect string
+		//and specializedshader_fragment becomes -> specializedshader_fragmenttt

```

Comment:
might be because the `IRR_CORE_UNIQUE_STRING_LITERAL_TYPE` macro only works for up to 127 character strings

Reply:
Length was 72

Reply:
still a problem?

Reply:
not a problem anymore